Build fails on i686: `error: ‘SO_REUSEPORT’ undeclared`

Build failed against `release-1.18.3` tag on i686. Reproducible in `satmandu/crew-pre-glibc-standalone:alex-i686.m58` docker container.

```
-- Found Git: /usr/local/bin/git (found version "2.53.0")
-- Setting package-version to 1.18.3
-- Setting ixml so-version to 11.1.8
-- Setting upnp so-version to 20.0.3
-- package-version is 1.18.3
-- The C compiler identification is GNU 15.2.0
-- The CXX compiler identification is GNU 15.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/local/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/local/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for sys/socket.h
-- Looking for sys/socket.h - found
-- Looking for ws2tcpip.h
-- Looking for ws2tcpip.h - not found
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of socklen_t
-- Check size of socklen_t - done
-- Check size of off_t
-- Check size of off_t - done
-- Check size of off_t
-- Check size of off_t - done
-- _FILE_OFFSET_BITS=64 needed
-- Looking for fseeko
-- Looking for fseeko - found
-- Looking for strnlen
-- Looking for strnlen - found
-- Looking for strndup
-- Looking for strndup - found
-- Performing Test HAVE_MACRO_PREFIX_MAP
-- Performing Test HAVE_MACRO_PREFIX_MAP - Success
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE
-- Check size of pthread_rwlock_t
-- Check size of pthread_rwlock_t - done
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Configuring done (3.8s)
-- Generating done (0.0s)
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_LIBRARY_PATH


-- Build files have been written to: /usr/local/tmp/crew/pupnp.20260320081522.dir/builddir
ninja: Entering directory `./builddir'
ninja: Jobserver mode detected:  -j8 --jobserver-auth=fifo:/usr/local/tmp/crew/pupnp.20260320081522.dir/jobserver_pool
[65/167] Building C object upnp/CMakeFiles/upnp_shared.dir/src/ssdp/ssdp_device.c.o
FAILED: [code=1] upnp/CMakeFiles/upnp_shared.dir/src/ssdp/ssdp_device.c.o 
/usr/local/bin/cc -DLIBUPNP_EXPORTS -DNDEBUG -D_FILE_OFFSET_BITS=64 -Dupnp_shared_EXPORTS -I/usr/local/tmp/crew/pupnp.20260320081522.dir/upnp/src/threadutil -I/usr/local/tmp/crew/pupnp.20260320081522.dir/upnp/inc -I/usr/local/tmp/crew/pupnp.20260320081522.dir/upnp/src/inc -I/usr/local/tmp/crew/pupnp.20260320081522.dir/builddir -I/usr/local/tmp/crew/pupnp.20260320081522.dir/builddir/upnp/inc -I/usr/local/tmp/crew/pupnp.20260320081522.dir/ixml/inc -I/usr/local/tmp/crew/pupnp.20260320081522.dir/ixml/src/inc -O3 -pipe -ffat-lto-objects -fPIC  -flto=auto -flto=auto -O3 -DNDEBUG -flto=auto -fno-fat-lto-objects -fPIC -fvisibility=hidden -fmacro-prefix-map=/usr/local/tmp/crew/pupnp.20260320081522.dir/=. -pthread -MD -MT upnp/CMakeFiles/upnp_shared.dir/src/ssdp/ssdp_device.c.o -MF upnp/CMakeFiles/upnp_shared.dir/src/ssdp/ssdp_device.c.o.d -o upnp/CMakeFiles/upnp_shared.dir/src/ssdp/ssdp_device.c.o -c /usr/local/tmp/crew/pupnp.20260320081522.dir/upnp/src/ssdp/ssdp_device.c
/usr/local/tmp/crew/pupnp.20260320081522.dir/upnp/src/ssdp/ssdp_device.c: In function ‘SendToCaller’:
/usr/local/tmp/crew/pupnp.20260320081522.dir/upnp/src/ssdp/ssdp_device.c:270:17: error: ‘SO_REUSEPORT’ undeclared (first use in this function); did you mean ‘SO_REUSEADDR’?
  270 |                 SO_REUSEPORT,
      |                 ^~~~~~~~~~~~
      |                 SO_REUSEADDR
/usr/local/tmp/crew/pupnp.20260320081522.dir/upnp/src/ssdp/ssdp_device.c:270:17: note: each undeclared identifier is reported only once for each function it appears in
[72/167] Linking C shared library ixml/libixml.so.11.1.8
ninja: build stopped: subcommand failed.
```

--- a/upnp/src/ssdp/ssdp_device.c
+++ b/upnp/src/ssdp/ssdp_device.c
@@ -249,6 +249,7 @@ static int SendToCaller(
 		#endif
 	char buf_ntop[INET6_ADDRSTRLEN];
 	int ret = UPNP_E_SUCCESS;
+	int bindrc = 0;
 
 	ReplySock = socket(res->ai_family, res->ai_socktype, res->ai_protocol);
 	if (ReplySock == INVALID_SOCKET) {
@@ -264,7 +265,8 @@ static int SendToCaller(
 	PROCESS_SOCKET_ERROR(
 		__FILE__, __LINE__, rc, UPNP_E_SOCKET_ERROR, "setsockopt-1");
 		#if (defined(BSD) && !defined(__GNU__)) || \
-			defined(__APPLE__) || defined(__linux__)
+			defined(__APPLE__) || \
+			(defined(__linux__) && defined(SO_REUSEPORT))
 	rc = setsockopt(ReplySock,
 		SOL_SOCKET,
 		SO_REUSEPORT,
@@ -272,8 +274,9 @@ static int SendToCaller(
 		sizeof(yes));
 	PROCESS_SOCKET_ERROR(
 		__FILE__, __LINE__, rc, UPNP_E_SOCKET_ERROR, "setsockopt-1x");
-		#endif /* BSD, __APPLE__, __linux__ */
-	ssize_t bindrc = bind(ReplySock, res->ai_addr, res->ai_addrlen);
+		#endif /* (BSD && !__GNU__) || __APPLE__ || (__linux__ && \
+			  SO_REUSEPORT) */
+	bindrc = bind(ReplySock, res->ai_addr, res->ai_addrlen);
 	PROCESS_SOCKET_ERROR(
 		__FILE__, __LINE__, bindrc, UPNP_E_SOCKET_BIND, "bind");
 

--- a/upnp/src/ssdp/ssdp_device.c
+++ b/upnp/src/ssdp/ssdp_device.c
@@ -909,8 +909,8 @@ int DeviceAdvertisement(char *DevType,
 		ret_code = UPNP_E_INVALID_PARAM;
 		goto error_handler;
 	}
-	/* If deviceis a root device , here we need to send 3 advertisement
-	 * or reply */
+	/* If device is a root device, here we need to send 3 advertisements or
+	 * reply */
 	if (RootDev) {
 		rc = snprintf(
 			Mil_Usn, sizeof(Mil_Usn), "%s::upnp:rootdevice", Udn);

